_posts/2021-03-05-automatically-generated-api-tests-with-clojure-and-reitit.md

```diff
@@ -0,0 +1,332 @@
+---
+layout: post
+title: Automatically generated API tests with Clojure and Reitit
+author: jgke
+excerpt: >
+  Is your entire API tested? Maybe you should test that.
+tags:
+ - Clojure
+ - Reitit
+ - Testing
+---
+
+Have you ever forgotten to write a test for some API endpoint, which then
+eventually happens to break? Would it be possible to check whether all of your
+API is tested? If you are using a data-driven router, it's possible to write a
+test which walks over the entire API.
+
+My current customer project is using something along these lines to test that
+the entire API handles credentials and permissions appropriately and doesn't
+randomly crash at some endpoints. As a consequence, the tests generated by
+these methods happen to create an almost complete smoke test suite, where a new
```

Comment:
What does "these methods" refer to? Is there one test or more than one? Previously, you write "test", but here it's "tests".

If it's just one test, maybe something like:

>It also happens to create an almost complete smoke test suite…

Does it take "a few lines of code" to add a new endpoint, or to add a test for a new endpoint?